src/api/bulkQuotes/{id}/error.js

```diff
@@ -52,52 +52,30 @@ module.exports = {
      * responses: 200, 400, 401, 403, 404, 405, 406, 501, 503
      */
   put: async function BulkQuotesErrorById (context, request, h) {
+    // todo: the same as PUT /bulkQuotes
     const histTimerEnd = Metrics.getHistogram(
-      'quotes_id_get',
+      'bulkQuotes_id_put_error',
       'Process HTTP PUT /bulkQuotes/{id}/error request',
       ['success']
     ).startTimer()
-    // log request
-    Logger.isDebugEnabled && Logger.debug(`got a PUT /bulkQuotes/{id}/error request: ${util.inspect(request.payload)}`)
 
-    // instantiate a new quote model
-    const model = new BulkQuotesModel({
-      db: request.server.app.database,
-      requestId: request.info.id
-    })
+    try {
+      logger.debug('got a PUT /bulkQuotes error request: ', request.payload)
+      const { topic, config } = kafkaConfig.PRODUCER.BULK_QUOTE.PUT
+      const topicConfig = dto.topicConfigDto({ topicName: topic })
+      const message = dto.messageFromRequestDto(request, Events.Event.Type.BULK_QUOTE, Events.Event.Action.PUT)
 
-    const quoteRequest = {
-      payload: { ...request.payload },
-      headers: { ...request.headers },
-      span: request.span,
-      params: { ...request.params }
-    }
+      await Producer.produceMessage(message, topicConfig, config)
 
-    // extract some things from the request we may need if we have to deal with an error e.g. the
-    // originator and quoteId
-    const bulkQuoteId = quoteRequest.params.id
-    const fspiopSource = quoteRequest.headers[Enum.Http.Headers.FSPIOP.SOURCE]
-    const span = quoteRequest.span
-    try {
-      const spanTags = LibUtil.getSpanTags(quoteRequest, Enum.Events.Event.Type.BULK_QUOTE, Enum.Events.Event.Action.ABORT)
-      span.setTags(spanTags)
-      await span.audit({
-        headers: quoteRequest.headers,
-        payload: quoteRequest.payload
-      }, EventSdk.AuditEventAction.start)
-      // call the quote error handler in the model
-      model.handleBulkQuoteError(quoteRequest.headers, bulkQuoteId, quoteRequest.payload.errorInformation, span).catch(err => {
-        Logger.isErrorEnabled && Logger.error(`ERROR - handleBulkQuoteError: ${LibUtil.getStackOrInspect(err)}`)
-      })
       histTimerEnd({ success: true })
     } catch (err) {
-      // something went wrong, use the model to handle the error in a sensible way
-      Logger.isErrorEnabled && Logger.error(`ERROR - PUT /bulkQuotes/{id}/error: ${LibUtil.getStackOrInspect(err)}`)
-      model.handleException(fspiopSource, bulkQuoteId, err, quoteRequest.headers)
+      logger.error(`error in PUT /bulkQuotes error request: ${err?.message}`)
+      // todo: think, how we should handle such error cases:
+      //   - how to send callback ?
+      //   - OR reply with errorCode (not 200)?
```

Comment:
To be honest, I wasn't sure how it's better to handle such errors during publishing a message. But if you think we can use _model.handleException_ here, I'll update these parts of API handlers

Reply:
Reposting here since I believe this is how the ml-api-adapter handles it.

Believe we would just rethrow the error and have a http syncronous 500 be sent back if something goes wrong publishing the message.
The kafka handler should handle the sending of the error callback notification if something goes wrong processing wise.

Reply:
@vijayg10 , what do you think regarding Kevin's suggestion?  Let's make a final decision in which way I should handle publishing error